lib/puppet/file_system.rb

```diff
@@ -122,8 +122,8 @@ def self.each_line(path, &block)
   #
   # @api public
   #
-  def self.read(path)
-    @impl.read(assert_path(path))
+  def self.read(path, opts = {})
+    @impl.read(assert_path(path), opts)
```

Comment:
Should we warn if `:encoding` is not specified? If it's missing, the caller is likely confusing byte vs character streams. And if they really do want a byte stream, then they could specify as `:encoding => ascii_8bit`.

Or we could add a `read_characters` method that requires an `:encoding`?

It's times like these where I miss Java's InputStreams and Readers.